py/server/deephaven/arrow.py

```diff
@@ -0,0 +1,148 @@
+#
+#     Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+"""This module supports conversions between Arrow tables and Deephaven tables."""
+
+from typing import List, Dict
+
+import jpy
+import pyarrow as pa
+
+from deephaven import DHError
+from deephaven.table import Table
+
+_JArrowToTableConverter = jpy.get_type("io.deephaven.extensions.barrage.util.ArrowToTableConverter")
+_JTableToArrowConverter = jpy.get_type("io.deephaven.extensions.barrage.util.TableToArrowConverter")
+
+_ARROW_DH_DATA_TYPE_MAPPING = {
+    pa.null(): '',
+    pa.bool_(): 'java.lang.Boolean',
+    pa.int8(): 'byte',
+    pa.int16(): 'short',
+    pa.int32(): 'int',
+    pa.int64(): 'long',
+    pa.uint8(): '',
+    pa.uint16(): 'char',
+    pa.uint32(): '',
+    pa.uint64(): '',
+    pa.float16(): '',
+    pa.float32(): 'float',
+    pa.float64(): 'double',
+    pa.time32('s'): '',
+    pa.time32('ms'): '',
+    pa.time64('us'): '',
+    pa.time64('ns'): 'io.deephaven.time.DateTime',
+    pa.timestamp('s', tz=None): '',
+    pa.timestamp('ms', tz=None): '',
+    pa.timestamp('us', tz=None): '',
+    pa.timestamp('ns', tz=None): '',
```

Comment:
As noted later, it does look like we support timestamps if a timezone is present.

Reply:
It's possible that we only support the `ns` resolution version of this timestamp.

Reply:
This looks like it is covered in https://github.com/deephaven/deephaven-core/issues/3223